### ISSUE TITLE
I want to use FocalLossV1,but has the error that your loss has no attribute to backward().

### ISSUE THREAD

**Issue by hanna6666** [CoinCheung/pytorch-loss#12]
Hi，I want to use FocalLossV1,but has the error that your loss has no attribute to backward().
How should I use it in right way?

**Comment by CoinCheung** [comment]
Would you paste a piece of code of how you triggered the error?

**Comment by hanna6666** [comment]
`def train(epoch):
    print('\nEpoch: %d' % epoch)
    net.train()
    train_loss = 0
    correct = 0
    total = 0
    for batch_idx, (inputs, targets) in enumerate(trainloader):
        inputs, targets = inputs.to(device), targets.to(device)
        optimizer.zero_grad()
        outputs = net(inputs)
        # loss = criterion(outputs, targets)
        loss = FocalLossV1(outputs,targets)
        print(loss)
        loss.backward()
        optimizer.step()

        train_loss += loss.item()
        train_loss += loss
        _, predicted = outputs.max(1)
        total += targets.size(0)
        correct += predicted.eq(targets).sum().item()

        progress_bar(batch_idx, len(trainloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
                     % (train_loss/(batch_idx+1), 100.*correct/total, correct, total))

    return train_loss/(batch_idx+1)
`

**Comment by hanna6666** [comment]
Here is  my train code

`def train(epoch):
    print('\nEpoch: %d' % epoch)
    net.train()
    train_loss = 0
    correct = 0
    total = 0
    for batch_idx, (inputs, targets) in enumerate(trainloader):
        inputs, targets = inputs.to(device), targets.to(device)
        optimizer.zero_grad()
        outputs = net(inputs)
        loss = FocalLossV1(outputs,targets)
        loss.backward()
        optimizer.step()

        train_loss += loss.item()
        train_loss += loss
        _, predicted = outputs.max(1)
        total += targets.size(0)
        correct += predicted.eq(targets).sum().item()

        progress_bar(batch_idx, len(trainloader), 'Loss: %.3f | Acc: %.3f%% (%d/%d)'
                     % (train_loss/(batch_idx+1), 100.*correct/total, correct, total))

    return train_loss/(batch_idx+1)
`

**Comment by CoinCheung** [comment]
Are you using the class of `FocalLossV1` directly ? Please try this:
```
crit = FocalLossV1()
...
loss = crit(outputs, targets)
loss.backward()
...
```

If I did not understand your problem well, please paste the error message.

**Comment by hanna6666** [comment]
it worked! Thank you so much!!!!!!!

**Comment by CoinCheung** [comment]
I am closing this because the problem is solved.
